### PR TITLE
Add draft of document describing pull request requirements

### PR DESCRIPTION
**Pull request by tituomin**

### CODE REVIEW

**Review by petterill — COMMENTED** [review]
This is good stuff but 
- the DoD section should be merged with https://github.com/City-of-Helsinki/bestpractice/blob/master/docs/scrum-project-management.md#definition-of-done
- The Finnish language section is probably not relevant anymore, as the language for the site seems to be english only (although I don't think there has been a decision on writinh in english exclusively)

**Review by pkalliok — CHANGES_REQUESTED** [review]
This should really be merged with the DoD in scrum-project-management.md

File: dod.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      * When doing public API functionality context of the code is written
+      * Clarification, when code might be unclear.
+      * Clear and well named functionality does not need a comment
+- [ ] Variable names convey their use case
+- [ ] Produced code matches presumed functionality
+      * Matches specifications given in the issue
+      * Works in desired environments
+- [ ] Code quality
+      * Code follows best practices
+        * Usually specified by the language, framework or library itself.
```
Comment (pkalliok) [inline dod.md:29]: These I think belong to coding standards.

File: dod.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      * Code follows styleguide
+      * Project specific linter rules
+        * If no linting rules are specified each project should follow some kind of guidelines for unified code
+- [ ] Functionality verified by testing the application
+- [ ] Each function and class should have a single clear purpose
+      * Complex or long function sections should be split into smaller ones
+      * Each additional **if**, **loop**, **switch** and **logical conditional operation** adds complexity
+      * Complexity can be measured by this way if each complexity point is added together.
```
Comment (pkalliok) [inline dod.md:37]: these too.

**Review by pkalliok — COMMENTED** [review]
Some other stuff to move to coding standards.

File: dod.md (hunk 1 above)
Comment (tituomin) [reply dod.md:29]: "code follows the coding standard document [link]" ?

**Comment by petterill** [comment]
Closing this as it has been stale for so long. Can be reopened if something happens to this.

**Pull request closed without merging**